src/OrchardCore/OrchardCore.Media.Abstractions/Indexing/MediaFileIndexingOptions.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OrchardCore.Media.Indexing
+{
+    public class MediaFileIndexingOptions
+    {
+        private readonly Dictionary<string, Type> _mediaFileTextProviderRegistrations = new Dictionary<string, Type>();
+
+        public MediaFileIndexingOptions RegisterMediaFileTextProvider<TMediaFileTextProvider>(string fileExtension)
+            where TMediaFileTextProvider : class, IMediaFileTextProvider
+        {
+            return RegisterMediaFileTextProvider(fileExtension, typeof(TMediaFileTextProvider));
+        }
+
+        public MediaFileIndexingOptions RegisterMediaFileTextProvider(string fileExtension, Type providerType)
+        {
+            // Deliberate overwrite behavior so you can override registrations.
+            _mediaFileTextProviderRegistrations[NormalizeFileExtension(fileExtension)] = providerType;
+
+            return this;
+        }
+
+        public Type GetRegisteredMediaFileTextProvider(string fileExtension)
+        {
+            if (_mediaFileTextProviderRegistrations.TryGetValue(NormalizeFileExtension(fileExtension), out var providerType))
+            {
+                return providerType;
+            }
+
+            return null;
+        }
+
+        public bool AnyMediaFileTextProviders()
```

Comment:
What we normally do is just expose the 
`private readonly Dictionary<string, Type> _mediaFileTextProviderRegistrations = new Dictionary<string, Type>();`

as a readonly dictionary, so we don't need all these extra methods, for `Any()` etc.

so

`public IReadOnlyDictionary<string, Type> MediaFileTextProviders => _mediaFileTextProviders`

but up to you

Reply:
I'd hide that implementation detail.